src/js/tech/tech.js

```diff
@@ -204,7 +118,40 @@ class Tech extends Component {
       if (bufferedPercent === 1) {
         this.stopTrackingProgress();
       }
-    }, 500);
+    }), 500);
+  }
+
+  onDurationChange() {
+    this.duration_ = this.duration();
+  }
+
+  bufferedPercent() {
```

Comment:
This is still duplicated logic from the player I think. We should figure out the right way to share that. It could be just having the player ask the tech for buffered percent instead  of calculating itself.